clients/src/main/java/org/apache/kafka/common/resource/ResourceType.java

```diff
@@ -104,17 +104,18 @@ public static ResourceType fromCode(byte code) {
         return resourceType;
     }
 
-    private final byte code;
+    //making this int since profiling showed miss alignement.
+    private final int code;
```

Comment:
This improved the performance because the byte was not aligned in memory resulting in a slow memory access path.